### PR TITLE
First implementation of spare capacity management

### PR DESCRIPTION
**Pull request by io-debug**
This commit contains the first implementation to handle the spare capacity ("Notstromreserve").

**_Features**
-Read the current spare capacity over the Web-API implementation
-Change/set the spare capacity over the Web-API implementation 

To do so, just add the new sensor to a dashboard an use the provided slider to update the spare capacity.
Please note: The new sensor is deactivated by default, since it's still under development (please see "state of development" below)

**_Changes**
-Implementation of a the "number" plattform
-Added a number-sensor
-Added the needed requests to the Web-API

**_State of development**
Reading and updating the spare capacity works.
Nevertheless the shown spare capacity in the dashboard is not updated directly after you changed it. The change will just be shown, after the next update intervall of the Web-API.
Here I still have to find a way, how to force an update, that the chance is visible directly.
Happy, if you should have an idea. :-)

### CODE REVIEW

File: custom_components/senec/number.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        #self.entity_id = f"number.{slugify(title)}_{key}"
+        #self._attr_name = f"{title} {name}"
+
+
+
+    @property
+    def state(self) -> int:
+        number = self.entity_description.key
+        value = getattr(self.coordinator.senec, number)
+        return float(value)
```
Comment (marq24) [inline custom_components/senec/number.py:57]: hopefully we can use here int(value) ;-)

File: custom_components/senec/pysenec_ha/__init__.py [hunk 2]
```diff
@@ -1602,6 +1658,11 @@ async def update_get_systems(self, a_plant_number: int):
                 self._isAuthenticated = False
                 await self.authenticate(doUpdate=False, throw401=False)
 
+    @property
+    def spare_capacity(self) -> float:
+        if hasattr(self, '_spare_capacity'):
+            return str(self._spare_capacity)
```
Comment (marq24) [inline custom_components/senec/pysenec_ha/__init__.py:1664]: here I would use again int(self._spare_capacity) and also use the return type int (and not float)

File: custom_components/senec/const.py [hunk 3]
```diff
@@ -75,6 +79,21 @@ class ExtSensorEntityDescription(SensorEntityDescription):
 class ExtBinarySensorEntityDescription(BinarySensorEntityDescription):
     icon_off: str | None = None
 
+"""Supported number implementations"""
+NUMBER_SENSOR_TYPES = [
```
Comment (marq24) [inline custom_components/senec/const.py:83]: can we call this WEB_NUMBER_SENYOR_TYPES (so we have always the info, what integration the constants are valid for)

**Comment by marq24** [comment]
Thank you very much - beside really few comment's the change seams to be a great addon...

I would ask you, 1) to add yourself to the 'codeowners' list 2) also include an updated readme.md file in the PR that explains the functionality (since I assume, this requires on of the "Notstrom"Packages (which I don't have)...

File: custom_components/senec/number.py (hunk 1 above)
Comment (io-debug) [reply custom_components/senec/number.py:57]: Thanks for the review, I guess it was to late yesterday. :-) 
Updated and successfully tested.

File: custom_components/senec/pysenec_ha/__init__.py (hunk 2 above)
Comment (io-debug) [reply custom_components/senec/pysenec_ha/__init__.py:1664]: Updated and successfully tested.

File: custom_components/senec/const.py (hunk 3 above)
Comment (io-debug) [reply custom_components/senec/const.py:83]: Changed to WEB_NUMBER_SENYOR_TYPES

**Comment by io-debug** [comment]
Code changes done.
I'll update the readme.md in the next minutes.

**Comment by io-debug** [comment]
So, all changes are now commited.
Also the integration now updates the data, after a user changed the spare capacity. So a dashboard shows the changed spare capacity immediately.

**Pull request merged**